libs/queue/src/main.sw

```diff
@@ -0,0 +1,98 @@
+// The `Queue` type corresponds to the same data structure.
```

Comment:
This _comment_ does not inform me of what a Queue type is or what the data structure may be. Moreover, it's a comment (uses 2 forward slashes) rather than a documentation comment (uses 3 forward slashes).
I would elaborate on what a queue is for those who have never heard of such a type.

Reply:
Added module level docs.